### ISSUE TITLE
Pass keys instead of tokens

### ISSUE THREAD

**Issue by dk4210** [twilio/twilio-php#747]
Does this package support passing API keys instead of the token?

**Comment by Hunga1** [comment]
Yes, the SDK supports using API keys instead of the token. Please refer to the following doc, which includes code snippets for Ruby:

https://www.twilio.com/docs/iam/keys/api-key

**Comment by dk4210** [comment]
So I would install your package and then use the PHP code that Twillo provides? Is that correct?

**Comment by Hunga1** [comment]
Yes. The linked document can be used to get you started using API keys. You can generate API keys via code, or alternatively in the Twilio Console, both are highlighted in the doc.

**Comment by dk4210** [comment]
Hey one more question

I'm trying to use my account "main" key to generate future keys instead of using the account SID and account token.

I logged into my Twilio account and generated a new token with the key type as "main"


Normally I generate a new key like this. This process works, but I want to use a "Master" key to generate new ones.

 ```
               // Generate the key
                $sid =  "account sid";
                $token = "account token";
                $twilio = new Twilio($sid, $token);
                $new_key = $twilio->newKeys->create(["friendlyName" => "Some name"]);

                /* For testing */
                echo "<br>This is the new key that was generated - " . $new_key->sid;
                echo "<br>This is the new key secret - " . $new_key->secret;

```
I would like to use the new "main" key that I generated instead.

 ```
               // Generate the key
                $sid =  "token sid";
                $token = "token secret";
                $twilio = new Twilio($sid, $token);
                $new_key = $twilio->newKeys->create(["friendlyName" => "Some name"]);

                /* For testing */
                echo "<br>This is the new key that was generated - " . $new_key->sid;
                echo "<br>This is the new key secret - " . $new_key->secret;

```
When I try to generate a key, I receive the following error

```
"Twilio Exceptions RestException

[HTTP 400] Unable to create record: The request body violates the API contract
```

Any idea how to accomplish this?

Thanks!

**Comment by dk4210** [comment]
I figured it out. In order to use a "main" key to generate new keys, you have to create a sub account first.
Generate a main key in there and use the SID of that account, plus the key sid & secret

` // Generate the key (Sub Account)
                $sid =  "Sub account sid";
                $apiKey = "sub account main key";
                $apiSecret = "key secret";
                $twilio = new Twilio($apiKey, $apiSecret, $sid);
               $new_key = $twilio->newKeys->create(["friendlyName" => "Some name"]);

                /* For testing */
                echo "<br>This is the new key that was generated - " . $new_key->sid;
                echo "<br>This is the new key secret - " . $new_key->secret;
`

Hope this helps someone.  :-)
